sky/execution.py

```diff
@@ -972,27 +983,78 @@ def serve_up(
     Args:
         task: sky.Task to serve up.
         service_name: Name of the service.
-        controller_resources: The resources requirement for the controller.
-        controller_best_resources: The optimized resources for the controller.
     """
-    controller_cluster_name = serve.generate_controller_cluster_name(
-        service_name)
-    controller_best_resources.cloud.check_cluster_name_is_valid(
-        controller_cluster_name)
+    if re.fullmatch(serve.SERVICE_NAME_VALID_REGEX, service_name) is None:
```

Comment:
reuse `check_cluster_name_is_valid` function?

Reply:
One problem is that the error message printed by the function says that `cluster name is invalid`, which is a little bit confusing... Reference:
https://github.com/skypilot-org/skypilot/blob/e164dc29fc9aee2b14c4abe634af9b3b84e9a806/sky/clouds/cloud.py#L478-L498

Reply:
Shall we modify its name or add some flags to print `Service name is invalid`?